### PR TITLE
Add error message to valuation of zero.

### LINKED ISSUE #982: valuation(0,2) returns 0
```
julia> valuation(0,2)
0
```
This is a wrong answer. It should raise an error instead (or return -Inf).

### PR DESCRIPTION
**Pull request by wbhart**
Fixes #982

### CODE REVIEW

File: src/flint/qadic.jl [hunk 1]
```diff
@@ -753,7 +753,7 @@ function (R::FlintQadicField)(a::Int)
 end
 
 function (R::FlintQadicField)(n::fmpz)
-   if isone(n)
+   if iszero(n) || isone(n)
       N = 0
```
Comment (thofma) [inline src/flint/qadic.jl:757]: Unrelated?

File: src/flint/fmpz.jl [hunk 2]
```diff
@@ -1378,6 +1378,7 @@ remove(x::Integer, y::Integer) = remove(fmpz(x), fmpz(y))
 Return the largest $n$ such that $y^n$ divides $x$.
 """
 function valuation(x::fmpz, y::fmpz)
+   x == 0 && error("Not yet implemented")
```
Comment (thofma) [inline src/flint/fmpz.jl:1381]:
```suggestion
   iszero(x) && error("Not yet implemented")
```

File: src/flint/qadic.jl (hunk 1 above)
Comment (wbhart) [reply src/flint/qadic.jl:757]: No, the qadic code fails with the other change without this.

Comment (thofma) [reply src/flint/qadic.jl:757]: OK
